content/tutorials/database_conventions/index.md

```diff
@@ -0,0 +1,447 @@
+  * Clustered indexes are created on DATA filegroup; non-clustered indexes on INDEXES filegroup.
+  * Leave the rest of the indexing up to the DBA. In order to help the DBA in choosing the correct indexes, explain him the (search and sorting) behaviour of the application.
+  * Performance issues might require the creation (or deletion) of indexes during the lifetime of the db. Indexes do not contribute to the versioning of the database.
+  * Naming: IN\_\<table name\>\_\<specification\> where specification describes function of index or columns included
+   * IN\_Measurement\_MeasurementStatusID for an index on the MeasurementStatusID column of the Measurement table.
+  * This example creates an index on the SpeciesID column of the Measurement table. Non-clustered indexes are always created on the INDEXES filegroup:
+
+```
+CREATE NONCLUSTERED INDEX IN_Measurement_SpeciesID ON Measurement(SpeciesID) ON INDEXES;
+```
+
+  * Example on how to enforce the primary key using a clustered index. Clustered indexes ares always created on the DATA filegroup:
+
+```
+ALTER TABLE Measurement ADD     
+   CONSTRAINT PK_Measurement PRIMARY KEY CLUSTERED (ID) ON DATA;
+```
+
+# Views
+  * Stored queries.
+  * Views can be updatable.
+  * For creation of indexed views (e.g. on calculated columns)  consult the DBA.
+  * Don't use ORDER BY in views.
+  * Naming: VW\_\<logical name\>, e.g.
+    * VW\_ActiveMesearumentStatus, retrieves the currently active measurement statusses 
+
+# Synonyms
+  * Do not use synonyms; they get the databases intertwined.
+
+# Stored procedures
+  * no guidelines yet
+  
+# Functions
+  * no guidelines yet
+  
+# Sequences
+  * Use IDENTITY column instead of sequence to get an auto-increment behaviour at the table level.
+  * Use sequence to get an auto-increment behaviour across multiple tables (rare).
+  * Naming: SQ\_\<logical name\>
+
+# Security
+  * DBA's task to implement security.
+  * Good to know how it's normally implemented.
+  * Permissions are granted to database roles and users are assigned to database roles; the user gets the permissions of the database role to which it is assigned.
+  * Permissions are never granted to users
+  * When possible a user should be assigned to 1 database role only
+
+# Database Roles
+  * following custom database roles per database:
+    * DR\_Admin: has dba permissions on the database; dba can be implement this using following steps
+      * grant create permissions for all objects the dba wants to allow to be created (at database level), eg create table
+      * + grant alter any schema (at database level)
+    * DR\_\<schema\>\_Admin: has dba permissions on the schema <schema> of the database; dba can be implement this using following steps
+      * grant create permissions for all objects the dba wants to allow to be created (at database level), eg create table
+    * + grant alter (at \<schema\> level)
+    * DR\_Reader: has readonly (R) permissions on all tables and views
```

Comment:
How to handle sensitive data (personal information)? Such information should only be available on a need-to-know basis. e.g. someone coordinating a monitoring scheme might need access to personal information of the contributors. The analyst / statistician working with the data has no need for that kind of information.